deps/CMakeLists.txt

```diff
@@ -0,0 +1,67 @@
+# Copyright (c) Microsoft. All rights reserved.
+# Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+########## SDK Testing Dependencies ##########
+if (NOT ${use_installed_dependencies})
+
+    if (NOT TARGET azure_macro_utils_c AND EXISTS "${CMAKE_CURRENT_LIST_DIR}/azure-macro-utils-c/CMakeLists.txt")
+        add_subdirectory(${CMAKE_CURRENT_LIST_DIR}/azure-macro-utils-c)
+    endif()
+
+    if (${original_run_e2e_tests} OR ${original_run_unittests} OR ${run_sfc_tests})
+        if (NOT TARGET testrunnerswitcher)
+            add_subdirectory(${CMAKE_CURRENT_LIST_DIR}/azure-c-testrunnerswitcher)
+        endif()
+        if (NOT TARGET ctest)
+            # Get the repo if it's not there
+            add_subdirectory(${CMAKE_CURRENT_LIST_DIR}/azure-ctest)
+        endif()
+        enable_testing()
+    endif()
+
+
+    if (NOT TARGET umock_c)
+        # Get the repo if it's not there
+        add_subdirectory(${CMAKE_CURRENT_LIST_DIR}/umock-c)
+    endif()
+else()
+    if (NOT azure_macro_utils_cFOUND)
+        find_package(azure_macro_utils_c REQUIRED CONFIG)
+    endif ()
+    if (NOT umock_cFOUND)
+        find_package(umock_c REQUIRED CONFIG)
+    endif ()
+endif()
```

Comment:
Why do we not incude the uhttp in this section? 

Side thought: Are you considering moving utpm, umqtt and uamqp from the top level into deps/ ?  That seems more intuitive to me.  But may not be worth it for risk of backwards compatibility.

Reply:
![image](https://user-images.githubusercontent.com/36710865/146296292-ec205fda-7b55-42d0-82d4-feb62284cdf5.png)

In theory I think I agree with you but at this point a little less of a concern for me. Maybe in the future yes but you're right about the back compat. I'll likely leave for now.